src/apps/relay/ns_ioalib_engine_impl.c

```diff
@@ -3106,105 +3106,114 @@ int send_data_from_ioa_socket_nbh(ioa_socket_handle s, ioa_addr *dest_addr, ioa_
                                   int tos, int *skip) {
   int ret = -1;
 
+  if (!nbh)
+    return -1;
+
   if (!s) {
     ioa_network_buffer_delete(NULL, nbh);
     return -1;
   }
 
-  if (s->done || (s->fd == -1)) {
-    TURN_LOG_FUNC(TURN_LOG_LEVEL_INFO,
-                  "!!! %s: (1) Trying to send data from closed socket: 0x%lx (1): done=%d, fd=%d, st=%d, sat=%d\n",
-                  __FUNCTION__, (long)s, (int)s->done, (int)s->fd, s->st, s->sat);
-    TURN_LOG_FUNC(TURN_LOG_LEVEL_INFO, "!!! %s socket: 0x%lx was closed\n", __FUNCTION__, (long)s);
+  do {
```

Comment:
this is a weird way to use a do{}while() loop.

I think it would be better if you used a label + goto statements like most C code uses.